### PR TITLE
[network_traffic] Set map_to_ecs to true by default

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## Proposed commit message

See https://github.com/elastic/integrations/issues/8185

This sets `map_to_ecs` for all the `network_traffic` datastreams to true, in line with https://github.com/elastic/integrations/issues/8185.

I don't have a huge amount of context for this, so apologies if I forgot something. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by fearful-symmetry** [comment]
So, @efd6 I kind of want you to check my logic here. It looks like CI is failing on a handful of mappings:
```
test case failed: one or more errors found in documents stored in logs-network_traffic.mongodb-12810 data stream: [0] parsing field value failed: field "network_traffic.mongodb.cursorId" value "0" (float64): expected string or array of strings
[1] parsing field value failed: field "network_traffic.mongodb.startingFrom" value "0" (float64): expected string or array of strings
```

Looking at the packetbeat code, it looks like `cursorId` and `startingFrom` are supposed to be integers:
```
	m.event["cursorId"], err = d.readInt64()
	if err != nil {
		logp.Err("An error occurred while parsing OP_REPLY message: %s", err)
		return false, false
	}
	m.event["startingFrom"], err = d.readInt32()
	if err != nil {
		logp.Err("An error occurred while parsing OP_REPLY message: %s", err)
		return false, false
	}

```

However, I'm not sure why this wasn't caught earlier, and I'm also not sure if I'm just...allowed to change the types of ECS fields?

File: packages/network_traffic/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.32.0"
+  changes:
+    - description: Set `map_to_ecs` to enabled by default
+      type: enhancement
+      link: https://github.com/elastic/integrations/issues/10785
```
Comment (andrewkroh) [inline packages/network_traffic/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/10785
```

File: packages/network_traffic/data_stream/mongodb/fields/protocol.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@
         The number of documents in the reply.
 
     - name: startingFrom
-      type: keyword
+      type: long
```
Comment (andrewkroh) [inline packages/network_traffic/data_stream/mongodb/fields/protocol.yml:30]: I don't think we should change any of these from keyword to long. I think they are all keywords in standalone Packetbeat mappings too, right? Unless we know that range queries[^1] need to be supported, then I would keep keep them as is. To make the tests pass there are two solutions:

1. Convert numbers to strings via processors.
2. Add `numeric_keyword_fields` to the test configuration. (see https://github.com/elastic/elastic-package/blob/main/docs/howto/pipeline_testing.md)



[^1]: https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html#_which_type_should_i_use

Comment (fearful-symmetry) [reply packages/network_traffic/data_stream/mongodb/fields/protocol.yml:30]: So, annoyingly, the packetbeat fields don't seem to have types set?

```
        - name: startingFrom
          description: >
            Where in the cursor this reply is starting.
        - name: cursorId
          description: >
            The cursor identifier returned in the OP_REPLY. This must be the value that was returned from the database.

```

But the other fields do have the correct type set:

```
        - name: numberToReturn
          type: long
          description: >
            The requested maximum number of documents to be returned.

```

I wonder if that's how these got set to the wrong type to begin with? Either way, it seems like we won't be conflicting with any existing mapping? Maybe?

Comment (andrewkroh) [reply packages/network_traffic/data_stream/mongodb/fields/protocol.yml:30]: > So, annoyingly, the packetbeat fields don't seem to have types set?

The [default](https://github.com/elastic/beats/blob/9605c28053f0745dc63f7b99f6ab813fb46baed7/libbeat/template/processor.go#L112-L113) is `keyword` in Beats fields.yml framework. So these fields are keywords in Beats.

> I wonder if that's how these got set to the wrong type to begin with? 

I don't think `cursorId` is wrong. A `keyword` is generally the best choice for identifiers when numeric range queries are not needed.

There might be some use to having `startingFrom` be indexed as a long, but I wouldn't change from keyword to long without good reason like a feature request. These fields are associated with mongodb [opcodes](https://www.mongodb.com/docs/manual/legacy-opcodes/) that were deprecated in mongodb 5.0 from July 2021 (and that is now two majors ago), so doubt they are getting used much anyways.

I think adding `numeric_keyword_fields: [...]` to the test configuration to remove the strict check that _source contains strings is the best course of action to resolve the test failure.

**Review by andrewkroh — APPROVED** [review]

**Comment by efd6** [comment]
It looks to me like the sample events were not updated here.

**Pull request merged**